app/models/pet_application.rb

```diff
@@ -1,4 +1,5 @@
 class PetApplication < ApplicationRecord
   belongs_to :pet
   belongs_to :application
+  validates :pet_id, uniqueness: {scope: :application_id, message: "and Application combination must be unique"}
```

Comment:
Why do we need to validate pet_id followed by uniqueness?

Reply:
Because if an application tries to add the same pet, a PetApplication combination will exist multiple times for the same pet to application. This means you can add Buster infinite times because it shows up in the foreign key table that many times.